extensions/amp-youtube/0.1/amp-youtube.js

```diff
@@ -71,6 +91,58 @@ class AmpYoutube extends AMP.BaseElement {
     // the playback.
     return false;
   }
+
+  /** @private */
+  buildImagePlaceholder_() {
+    const imgPlaceholder = new Image();
+    const videoid = this.videoid_;
+
+    setStyles(imgPlaceholder, {
+      // Cover matches YouTube Player styling.
+      'object-fit': 'cover',
+      // Hiding the placeholder initially to give the browser time to fix
+      // the object-fit: cover.
+      'visibility': 'hidden'
+    });
+
+    // Because sddefault.jpg isn't available for all videos, we try to load
+    // it and fallback to hqdefault.jpg.
+    loadPromise(imgPlaceholder).then(() => {
+      // Poll until complete is set.
+      if (!imgPlaceholder.complete) {
+        return whenDocumentReady(this.getWin().document);
+      }
+    }).then(() => {
+      // A pretty ugly hack since onerror won't fire on YouTube image 404.
```

Comment:
> I'd assume it'd have the same behavior, but it might be worth trying.

Sorry, I meant to do a switch on `resp.ok` to test if it was a 404.

> It doesn't seem YouTube sets CORS headers so that wouldn't work, no?
 
Serves me right for trying this while looking at a youtube image in my browser. Doh.